src/main/java/org/jabref/gui/mergeentries/MergeEntriesHelper.java

```diff
@@ -0,0 +1,102 @@
+package org.jabref.gui.mergeentries;
+
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.jabref.gui.undo.NamedCompound;
+import org.jabref.gui.undo.UndoableChangeType;
+import org.jabref.gui.undo.UndoableFieldChange;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.FieldFactory;
+import org.jabref.model.entry.types.EntryType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for merging bibliography entries with undo support.
+ * Source entry data is merged into the library entry, with longer field values preferred
+ * and obsolete fields removed.
+ */
+public final class MergeEntriesHelper {
```

Comment:
There is no test for this. Please add a JUNit test (ctrl+shift+t in IntelliJ gets you started). -- Reading a unit test makes it easier to check whether the logic s correct.